Userland/Games/Flood/Board.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Board.h"
+#include <AK/Random.h>
+#include <time.h>
+
+Board::Board(size_t rows, size_t columns)
+{
+    resize(rows, columns);
+}
+
+Board::Board(Board const& board)
+    : m_rows(board.m_rows)
+    , m_columns(board.m_columns)
+    , m_current_color(board.m_current_color)
+    , m_previous_color(board.m_current_color)
+    , m_colors(board.m_colors)
+    , m_cells(board.m_cells)
+{
+}
+
+void Board::clear()
+{
+    for (size_t row = 0; row < m_rows; ++row) {
+        for (size_t column = 0; column < m_columns; ++column) {
+            set_cell(row, column, Color::Transparent);
+        }
+    }
+}
+
+bool Board::is_flooded() const
+{
+    Color first_cell_color = cell(0, 0).release_value();
+    for (size_t row = 0; row < rows(); ++row) {
+        for (size_t column = 0; column < columns(); ++column) {
+            if (cell(row, column).release_value() == first_cell_color)
+                continue;
+            return false;
+        }
+    }
+    return true;
+}
+
+void Board::randomize()
+{
+    for (size_t row = 0; row < m_rows; ++row) {
+        for (size_t column = 0; column < m_columns; ++column) {
+            Color const& color = m_colors[get_random<u32>() % m_colors.size()];
+            set_cell(row, column, color);
+        }
+    }
+    set_current_color(cell(0, 0).release_value());
+}
+
+void Board::resize(size_t rows, size_t columns)
+{
+    m_rows = rows;
+    m_columns = columns;
+
+    // Vector values get default-initialized, we don't need to set them explicitly.
+    m_cells.resize(rows);
+    for (size_t row = 0; row < rows; ++row)
+        m_cells[row].resize(columns);
+}
+
+void Board::set_cell(size_t row, size_t column, Color const& color)
+{
+    VERIFY(row < m_rows && column < m_columns);
+
+    m_cells[row][column] = color;
+}
+
+ErrorOr<Color> Board::cell(size_t row, size_t column) const
+{
+    if (row >= m_rows || column >= m_columns)
+        return Error::from_string_literal("No such cell.");
+
+    return m_cells[row][column];
+}
+
+void Board::set_current_color(Color new_color)
+{
+    m_previous_color = m_current_color;
+    m_current_color = new_color;
+}
+
+void Board::set_color_scheme(Vector<Color> const& colors)
+{
+    VERIFY(colors.size() == 8);
+    m_colors = colors;
+}
+
+void Board::reset()
+{
+    clear();
+    set_current_color(Color::Transparent);
+    m_previous_color = Color::Transparent;
+}
+
+// Adapted from Userland/PixelPaint/Tools/BucketTool.cpp::flood_fill.
+u32 Board::update_colors(bool only_calculate_flooded_area)
+{
+    Queue<Gfx::IntPoint> points_to_visit = Queue<Gfx::IntPoint>();
```

Comment:
You can either use `auto` on the left or just do
`Queue<Gfx::IntPoint> points_to_visit;` that should also default construct it